src/mscorlib/src/System/Text/ASCIIEncoding.cs

```diff
@@ -763,13 +763,46 @@ public override bool IsSingleByte
 
         public override Decoder GetDecoder()
         {
-            return new DecoderNLS(this);
+            return new ASCIIDecoder(this);
         }
 
-
         public override Encoder GetEncoder()
         {
-            return new EncoderNLS(this);
+            return new ASCIIEncoder(this);
+        }
```

Comment:
So now Encoding.ASCII.GetEncoder().AnyVirtual() ends up getting devirtualized all the way through?